### PR TITLE
[SYCL][NFC] Fix a misleading comment regarding the SYCL flow

### PR DESCRIPTION
**Pull request by AGindinson**
Signed-off-by: Artem Gindinson <artem.gindinson@intel.com>

### CODE REVIEW

File: clang/lib/Driver/ToolChains/Clang.cpp [hunk 1]
```diff
@@ -3944,10 +3944,13 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
   // Check number of inputs for sanity. We need at least one input.
   assert(Inputs.size() >= 1 && "Must have at least one input.");
   // CUDA/HIP compilation may have multiple inputs (source file + results of
-  // device-side compilations). OpenMP and SYCL device jobs also take the host
-  // IR as a second input. Module precompilation accepts a list of header files
-  // to include as part of the module. All other jobs are expected to have
-  // exactly one input.
+  // device-side compilations).
+  // OpenMP device jobs take the host IR as a second input.
+  // SYCL host jobs accept the integration header from the device-side
```
Comment (mdtoguchi) [inline clang/lib/Driver/ToolChains/Clang.cpp]: Should it be mentioned that the second input for SYCL is the integration header?

Comment (AGindinson) [reply clang/lib/Driver/ToolChains/Clang.cpp]: That was my intention, at least. Is it the new wording that causes the confusion?

Comment (mdtoguchi) [reply clang/lib/Driver/ToolChains/Clang.cpp]: Maybe something like `SYCL host jobs accept the integration header from the device-side compilation as the second input.`

Comment (AGindinson) [reply clang/lib/Driver/ToolChains/Clang.cpp]: Sounds good, updated

**Pull request merged**